File: examples/hello.lua
-- Lua Example - Lightweight scripting

print("🌙 Hello from Lua!")

-- Variables
local message = "Lua is lightweight and fast!"
local number = 42

print("Message: " .. message)
print("Number: " .. number)

-- Table (array)
local numbers = {1, 2, 3, 4, 5}
print("Numbers: " .. table.concat(numbers, ", "))

-- Function
function add(a, b)
    return a + b
end

local result = add(10, 20)
print("10 + 20 = " .. result)

-- Table (dictionary)
local person = {
    name = "John",
    age = 30,
    city = "New York"
}

print("\nPerson Info:")
for key, value in pairs(person) do
    print("  " .. key .. ": " .. value)
end

File: examples/hello.php
<?php
// PHP Example - Server-side scripting

echo "🐘 Hello from PHP!\n";

// Variables
$message = "PHP powers the web!";
$number = 42;

echo "Message: $message\n";
echo "Number: $number\n";

// Array
$numbers = [1, 2, 3, 4, 5];
echo "Numbers: " . implode(", ", $numbers) . "\n";

// Function
function add($a, $b) {
    return $a + $b;
}

$result = add(10, 20);
echo "10 + 20 = $result\n";

// Associative array
$person = [
    "name" => "John",
    "age" => 30,
    "city" => "New York"
];

echo "\nPerson Info:\n";
foreach ($person as $key => $value) {
    echo "  $key: $value\n";
}

?>
